doc/man/man3/EnvCmds.3
.TH "EnvCmds" 3 "Sat Jun 6 2015" "Version 0.0.1" "RIDE" \" -*- nroff -*-
.ad l
.nh
.SH NAME
EnvCmds \- 
.SH SYNOPSIS
.br
.PP
.SS "Functions"

.in +1c
.ti -1c
.RI "void \fBinitPtrs\fP ()"
.br
.ti -1c
.RI "void \fBsetPkgPathPtr\fP (QString *dirPathPtr)"
.br
.ti -1c
.RI "QString * \fBgetPkgPathPtr\fP ()"
.br
.ti -1c
.RI "void \fBsetArgsLstPtr\fP (QStringList *\fBargsLstPtr\fP)"
.br
.ti -1c
.RI "QStringList * \fBgetArgsLstPtr\fP ()"
.br
.ti -1c
.RI "void \fBgenRideProjConfigs\fP ()"
.br
.ti -1c
.RI "void \fBaddToPkgPtrVecPtr\fP (QDir *pkg)"
.br
.ti -1c
.RI "QDir * \fBsubFromPkgPtrVecPtr\fP (QDir *pkg)"
.br
.ti -1c
.RI "QVector< QDir * > * \fBgetPkgPtrVecPtr\fP ()"
.br
.ti -1c
.RI "bool \fBsourceEnv\fP ()"
.br
.ti -1c
.RI "bool \fBinitRosPkg\fP ()"
.br
.ti -1c
.RI "bool \fBtestRosPkg\fP ()"
.br
.ti -1c
.RI "bool \fBcatkin_make\fP ()"
.br
.in -1c
.SS "Variables"

.in +1c
.ti -1c
.RI "QString * \fBworkspacePathStrPtr\fP"
.br
.ti -1c
.RI "QString * \fBpkgPathStrPtr\fP"
.br
.ti -1c
.RI "QString * \fBlaunchFilePathStrPtr\fP"
.br
.ti -1c
.RI "QStringList * \fBargsLstPtr\fP"
.br
.ti -1c
.RI "QProcess * \fBprocessPtr\fP"
.br
.ti -1c
.RI "QVector< QDir * > * \fBpkgPtrVecPtr\fP"
.br
.ti -1c
.RI "bool \fBptrsWereInited\fP"
.br
.in -1c
.SH "Function Documentation"
.PP 
.SS "void EnvCmds::addToPkgPtrVecPtr (QDir *pkg)"

.SS "bool EnvCmds::catkin_make ()"

.PP
Definition at line 125 of file EnvCmds\&.cpp\&.
.SS "void EnvCmds::genRideProjConfigs ()"

.PP
Definition at line 38 of file EnvCmds\&.cpp\&.
.SS "QStringList * EnvCmds::getArgsLstPtr ()"

.PP
Definition at line 32 of file EnvCmds\&.cpp\&.
.SS "QString * EnvCmds::getPkgPathPtr ()"

.PP
Definition at line 20 of file EnvCmds\&.cpp\&.
.SS "QVector<QDir*>* EnvCmds::getPkgPtrVecPtr ()"

.SS "void EnvCmds::initPtrs ()"

.PP
Definition at line 4 of file EnvCmds\&.cpp\&.
.SS "bool EnvCmds::initRosPkg ()"

.PP
Definition at line 100 of file EnvCmds\&.cpp\&.
.SS "void EnvCmds::setArgsLstPtr (QStringList *argsLstPtr)"

.PP
Definition at line 26 of file EnvCmds\&.cpp\&.
.SS "void EnvCmds::setPkgPathPtr (QString *dirPathPtr)"

.PP
Definition at line 14 of file EnvCmds\&.cpp\&.
.SS "bool EnvCmds::sourceEnv ()"

.PP
Definition at line 91 of file EnvCmds\&.cpp\&.
.SS "QDir* EnvCmds::subFromPkgPtrVecPtr (QDir *pkg)"

.SS "bool EnvCmds::testRosPkg ()"

.PP
Definition at line 109 of file EnvCmds\&.cpp\&.
.SH "Variable Documentation"
.PP 
.SS "QStringList* EnvCmds::argsLstPtr"

.PP
Definition at line 31 of file EnvCmds\&.h\&.
.SS "QString* EnvCmds::launchFilePathStrPtr"

.PP
Definition at line 30 of file EnvCmds\&.h\&.
.SS "QString* EnvCmds::pkgPathStrPtr"

.PP
Definition at line 29 of file EnvCmds\&.h\&.
.SS "QVector<QDir*>* EnvCmds::pkgPtrVecPtr"

.PP
Definition at line 33 of file EnvCmds\&.h\&.
.SS "QProcess* EnvCmds::processPtr"

.PP
Definition at line 32 of file EnvCmds\&.h\&.
.SS "bool EnvCmds::ptrsWereInited"

.PP
Definition at line 34 of file EnvCmds\&.h\&.
.SS "QString* EnvCmds::workspacePathStrPtr"

.PP
Definition at line 28 of file EnvCmds\&.h\&.
.SH "Author"
.PP 
Generated automatically by Doxygen for RIDE from the source code\&.


src/EnvCmds.cpp
#include "EnvCmds.h"


void EnvCmds::initPtrs()
{
    pkgPathStrPtr = new QString();
    argsLstPtr = new QStringList();
    processPtr = new QProcess();

    pkgPtrVecPtr = new QVector<QDir*>();
}


void EnvCmds::setPkgPathPtr(QString* dirPathPtr)
{
    pkgPathStrPtr = dirPathPtr;
}


QString* EnvCmds::getPkgPathPtr()
{
    return pkgPathStrPtr;
}


void EnvCmds::setArgsLstPtr(QStringList* argsLstPtr)
{
    argsLstPtr = argsLstPtr;
}


QStringList* EnvCmds::getArgsLstPtr()
{
    return argsLstPtr;
}


void EnvCmds::genRideProjConfigs()
{
    //mkdir .rideProject //at catkin_ws/src/project_name
    //mkdir resources    //at .rideProject/
    //...
    QStringList* tmp = new QStringList();
    tmp->append(".rideProject");
    tmp->append(*getPkgPathPtr() );
    processPtr->execute("mkdir", *tmp);
    
    tmp->clear();
    tmp->append("resources");
    tmp->append(*getPkgPathPtr() + "/" + ".rideProject/");
    processPtr->execute("mkdir", *tmp);
    
    tmp->clear();
    tmp->append("project");
    tmp->append(*getPkgPathPtr() + "/" + ".rideProject/resources/");
    processPtr->execute("mkdir", *tmp);
    
    tmp->clear();
    tmp->append("configurations.xml");
    tmp->append(*getPkgPathPtr() + "/" + ".rideProject/resources/project/");
    processPtr->execute("touch", *tmp);
    
    tmp->clear();
    tmp->append("scout");
    tmp->append(*getPkgPathPtr() + "/" + ".rideProject/resources/");
    processPtr->execute("mkdir", *tmp);
    
    tmp->clear();
    tmp->append("libraries");
    tmp->append(*getPkgPathPtr() + "/" + ".rideProject/scout/");
    processPtr->execute("mkdir", *tmp);
    
    tmp->clear();
    tmp->append("project");
    tmp->append(*getPkgPathPtr() + "/" + ".rideProject/scout/");
    processPtr->execute("mkdir", *tmp);
    
    
    tmp->clear();
    tmp->append("scout.qrc");
    tmp->append(*getPkgPathPtr() + "/" + ".rideProject/scout/");
    processPtr->execute("touch", *tmp);



    
    
}


bool EnvCmds::sourceEnv()
{
    processPtr->setWorkingDirectory(*pkgPathStrPtr);
    processPtr->start("source", *(new QStringList("devel/setup.bash")));
    
    return true; // ***CODE STUB--REPLACE ME ***
}


bool EnvCmds::initRosPkg()
{
    processPtr->setWorkingDirectory(*pkgPathStrPtr);
    processPtr->start("catkin_create_pkg", *argsLstPtr);
    
    return true; // *** CODE STUB--REPLACE ME ***
}


bool EnvCmds::testRosPkg()
{
    if(processPtr->state() == QProcess::NotRunning)
    {
        processPtr->start("source", *(new QStringList("devel/setup.bash")));
        processPtr->start("catkin_make", *(new QStringList(argsLstPtr->at(0))));
    }
    else
    {
        cerr << "Sorry, but the process is busy creating the pkg" << endl;
    }

    return true; // *** CODE STUB--REPLACE ME ***
}


bool EnvCmds::catkin_make()
{
    processPtr->setWorkingDirectory(*pkgPathStrPtr);
    processPtr->execute("source", *(new QStringList("devel/setup.bash")));
    processPtr->start("catkin_make");
}